### PR TITLE
[inputstream] log error for use of deprecated inputstreamaddon property

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Log error for use of deprecated inputstreamaddon property so users know what to do to fix it.

## Motivation and Context

Since https://github.com/xbmc/xbmc/pull/18002 users would not know what causes their playback issues if using the old property.

## How Has This Been Tested?

OSX using ffmpegdirect.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by AlwinEsch — APPROVED** [review]
Makes absolutely sense to Log it :+1: 

Hard to know why a playback not works without, if old is used.

**Review by peak3d — APPROVED** [review]
Thx @phunkyfish !

File: xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp [hunk 1]
```diff
@@ -68,6 +68,15 @@ CInputStreamAddon::~CInputStreamAddon()
 
 bool CInputStreamAddon::Supports(BinaryAddonBasePtr& addonBase, const CFileItem &fileitem)
 {
+  // Error for users to show deprecation, can be removed in Kodi 20
```
Comment (enen92) [inline xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp:71]: Maybe add a doxygen `@deprecated` or even a `@todo` to make this easier to track and completely remove in v20?

Comment (phunkyfish) [reply xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp:71]: Makes sense, maybe both.

Comment (phunkyfish) [reply xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp:71]: Or just the @todo. I think the deprecation is for function calls.

**Pull request merged**